### ISSUE TITLE
Unbounded pixel range in RasterizeTriangle

### ISSUE THREAD

**Issue by originlake** [cdcseacave/openMVS#666]
Hello,

Recently we ran into an issue that the DensifyPointCloud seems to stall when processing some ill-posed data. It's OK to fail in processing this kind of data as the sfm reconstruction is bad, but if it stalls we have no way to tell it.

Some investigations show that the process spent a really long time running [RasterizeTriangle](https://github.com/cdcseacave/openMVS/blob/master/libs/Common/Types.inl#L2474). Then we checked the implementation of that part of codes, it seems minx, maxx, miny, maxy are unbounded. For the data like the image below, some triangle faces might have a very large area of coverage. For example, one face in my data created minx=-605388, maxx =33622, miny=-2857096, maxy=129400. 

Since the outbound pixels will still be rejected in the middle of the loop, should we limit their range at the beginning by something like below. I suppose this will also improve some processing speed to complex dataset.
```
minx = std::max<int>(minx, 0);
miny = std::max<int>(miny, 0);
maxx = std::min<int>(maxx, depthMap.cols);
maxy = std::min<int>(maxy, depthMap.rows);
```
![Screenshot from 2021-04-02 16-30-54](https://user-images.githubusercontent.com/11423220/113452140-695bbd80-93d1-11eb-820c-040ed1716480.png)
![Screenshot from 2021-04-02 16-32-23](https://user-images.githubusercontent.com/11423220/113452136-6791fa00-93d1-11eb-9131-ed0d73b929b8.png)

Thank you in advance

**Comment by cdcseacave** [comment]
in `develop` branch triangles rasterization was optimized in many ways, including bounds check like this, pls try it:
https://github.com/cdcseacave/openMVS/blob/464493cebbc21e544f389b49e0603681a8bf3170/libs/Common/Types.inl#L2612

**Comment by originlake** [comment]
Thanks, glad to see there is a better solution. Any plan to make it available for DensifyPointCloud, seems it still using the old approach.

**Comment by cdcseacave** [comment]
pls check the last develop, the problem should be corrected

**Comment by originlake** [comment]
It seems the method to get the K matrix is changed and it's different from the master branch. https://github.com/cdcseacave/openMVS/blob/b0059591e4c24fed646b1b34755def5eadc7b243/libs/MVS/Camera.h#L180
It works well after I revert that part, processed an ill-posed dataset with stable speed. 
I also processed a normal dataset, the final result looks decent. But I noticed it slows down a little bit comparing to master in this case, around 8% slower, I only tested one dataset, not sure if it's common. 
But it does help a lot, thanks again for your help.
